include/pstl_offload/internal/usm_memory_replacement.h

```diff
@@ -0,0 +1,245 @@
+// -*- C++ -*-
+//===----------------------------------------------------------------------===//
+//
+// Copyright (C) Intel Corporation
+//
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+// This file incorporates work covered by the following copyright and permission
+// notice:
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef _ONEDPL_PSTL_OFFLOAD_INTERNAL_USM_MEMORY_REPLACEMENT_H
+#define _ONEDPL_PSTL_OFFLOAD_INTERNAL_USM_MEMORY_REPLACEMENT_H
+
+#ifndef __SYCL_PSTL_OFFLOAD__
+#error "__SYCL_PSTL_OFFLOAD__ macro should be defined to include this header"
+#endif
+
+#include <atomic>
+#include <cstdlib>
+#include <cassert>
+#include <cerrno>
+
+#include <sycl/sycl.hpp>
+
+#include <oneapi/dpl/execution>
+
+#include "usm_memory_common.h"
+
+namespace oneapi {
+namespace dpl {
+namespace pstl_offload {
+
+static const auto& get_device_selector() {
```

Comment:
@Perhaps, we should replace that in favor of `static constexpr` variable and perhaps move is to `Holder`

Reply:
Done